### PR TITLE
optimize flag logic about wait

### PR DESCRIPTION
**Pull request by yxxhero**
optimize flag logic about wait

### CODE REVIEW

**Comment by yxxhero** [comment]
@mumoshu Regarding configuration priorities, I feel like we should make some decisions. It's a little unclear at the moment. WDYT?

**Comment by xiaomudk** [comment]
@yxxhero There was a lack of test cases here before, we should add them this time. WDYT?

**Comment by yxxhero** [comment]
@xiaomudk I think so.

File: pkg/state/helmx_test.go [hunk 1]
```diff
@@ -9,6 +9,130 @@ import (
...
+	tests := []struct {
+		name     string
+		release  *ReleaseSpec
+		syncOpts *SyncOpts
+		helmSpec HelmSpec
+		expected []string
+	}{
+		{
+			name:     "release wait for jobs",
+			release:  &ReleaseSpec{WaitForJobs: &[]bool{true, true}[0]},
```
Comment (xiaomudk) [inline pkg/state/helmx_test.go:22]: Just one `true`？

File: pkg/state/helmx_test.go [hunk 2]
```diff
@@ -9,6 +9,130 @@ import (
...
+		{
+			name:     "helm defaults wait for jobs",
+			release:  &ReleaseSpec{},
+			syncOpts: nil,
+			helmSpec: HelmSpec{WaitForJobs: true},
+			expected: []string{"--wait-for-jobs"},
+		},
+		{
+			name:     "release wait for jobs false",
+			release:  &ReleaseSpec{WaitForJobs: &[]bool{false, false}[0]},
```
Comment (xiaomudk) [inline pkg/state/helmx_test.go:43]: Just one `false`

File: pkg/state/helmx_test.go [hunk 3]
```diff
@@ -9,6 +9,130 @@ import (
...
+	tests := []struct {
+		name     string
+		release  *ReleaseSpec
+		syncOpts *SyncOpts
+		helmSpec HelmSpec
+		expected []string
+	}{
+		{
+			name:     "release wait",
+			release:  &ReleaseSpec{Wait: &[]bool{true, true}[0]},
```
Comment (xiaomudk) [inline pkg/state/helmx_test.go:84]: Just one

File: pkg/state/helmx_test.go [hunk 4]
```diff
@@ -9,6 +9,130 @@ import (
...
+		{
+			name:     "helm defaults wait",
+			release:  &ReleaseSpec{},
+			syncOpts: nil,
+			helmSpec: HelmSpec{Wait: true},
+			expected: []string{"--wait"},
+		},
+		{
+			name:     "release wait false",
+			release:  &ReleaseSpec{Wait: &[]bool{false, false}[0]},
```
Comment (xiaomudk) [inline pkg/state/helmx_test.go:105]: Just one

**Pull request merged**